src/main/java/core/basesyntax/strategy/mapvalidator/KeyAbsentValidator.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.strategy.mapvalidator;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public interface KeyAbsentValidator extends MapValidator {
+    @Override
+    default void validateMap(Map<String, Integer> fruitQuantityMap,
+                                    FruitTransaction transaction) {
+        String fruit = transaction.getFruit();
+        if (fruitQuantityMap.containsKey(fruit)) {
+            String exceptionMessage = "Fruit from transaction = ["
+                    + transaction + "] was already present in the map in quantity of = ["
+                    + fruitQuantityMap.get(fruit) + "]";
```

Comment:
Avoid hardcoding the exception message directly in the method. It would be better to use a constant for the exception message or construct the message dynamically within the exception itself to improve maintainability and readability.